buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			it("should error when the command fails", func() {
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-				if err := os.RemoveAll(platformDir); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-				if err, ok := err.(*buildpack.Error); !ok || err.Type != buildpack.ErrTypeBuildpack {
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			when("modifying the env fails", func() {
-				var appendErr error
-
-				it.Before(func() {
-					appendErr = errors.New("some error")
-				})
-
-				each(it, []func(){
-					func() {
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(appendErr)
-					},
-					func() {
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(appendErr)
-					},
-					func() {
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(appendErr)
-					},
-					func() {
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddRootDir(gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(nil)
-						mockEnv.EXPECT().AddEnvDir(gomock.Any(), gomock.Any()).Return(appendErr)
-					},
-				}, "should error", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkdir(t,
-						filepath.Join(appDir, "layers-A-v1", "layer1"),
-						filepath.Join(appDir, "layers-A-v1", "layer2"),
-					)
-					h.Mkfile(t, "[types]\n  build = true",
-						filepath.Join(appDir, "layers-A-v1", "layer1.toml"),
-						filepath.Join(appDir, "layers-A-v1", "layer2.toml"),
-					)
-					if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != appendErr {
-						t.Fatalf("Incorrect error: %s\n", err)
-					}
-				})
-			})
-
-			when("invalid unmet entries", func() {
-				when("missing name", func() {
-					it("should error", func() {
-						mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-						h.Mkfile(t,
-							"[[unmet]]\n",
-							filepath.Join(appDir, "build-A-v1.toml"),
-						)
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNotNil(t, err)
-						expected := "name is required"
-						h.AssertStringContains(t, err.Error(), expected)
-					})
-				})
-
-				when("invalid name", func() {
-					it("should error", func() {
-						mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L467-L477